addons/website/static/src/js/editor/widget_link.js

```diff
@@ -38,26 +44,54 @@ weWidgets.LinkTools.include({
             classes: {
                 "ui-autocomplete": 'o_website_ui_autocomplete'
             },
-        }
-        wUtils.autocompleteWithPages(this, this.$('input[name="url"]'), options);
+        };
+        this.$urlInput = this.$('input[name="url"]');
+        this.$('.o_we_link_action_change').toggle(this.$urlInput.val().includes('/web/'));
+        wUtils.autocompleteWithPages(this, this.$urlInput, options);
         this._adaptPageAnchor();
         return def;
     },
+    /**
+     * @returns {boolean} Whether or not link action is set to download
+     */
+    isDownloadEnable: function () {
+        return this.$link[0].dataset.linkAction === 'download';
+    },
+    /**
+     * If URL starts with /web/ or if the file name is included in URL, then file has been uploaded
+     *
+     * @returns {boolean} Whether the link points to a file or not
+     */
+    isFileUploaded: function () {
+        let url = this.$urlInput && this.$urlInput.val();
+        if (url === undefined) {
+            url = this.$link.attr('href');
+        }
+        let fileName = this.$link[0].dataset.fileName;
+        fileName = fileName && fileName.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+        return new RegExp(`${fileName ? fileName + '|' : ''}\\/web\\/`).test(url);
+    },
```

Comment:
Don't understand. And should be reviewed to not dependend of urlInput and/or link but only urlInput.